### PR TITLE
🌱 Add verifier action for pull requests

### LINKED ISSUE #4556: Use emoji for PRs like upstream
/kind feature

**Describe the solution you'd like**
Emoji titles can be useful to generate release notes quickly in categories; this change would also align CAPA with CAPI.

### PR DESCRIPTION
**Pull request by kranurag7**
This commit adds a verifier action that will check the title of the pull request.

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4556 

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Added verifier for action for pull request
```

### CODE REVIEW

**Comment by Ankitasw** [comment]
@kranurag7 please update the release note.

File: .github/workflows/pr-verify.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    types: [opened, edited, synchronize, reopened]
+
+jobs:
+  verify:
+    runs-on: ubuntu-latest
+    name: verify PR contents
+    steps:
+    - name: Verifier action
+      id: verifier
+      uses: kubernetes-sigs/kubebuilder-release-tools@3c3411345eedc489d1022288aa844691e92a9c29 # tag=v0.4.2
```
Comment (richardcase) [inline .github/workflows/pr-verify.yml:14]: If we are going to require the use of emojis then we need to update the PR template:

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/.github/PULL_REQUEST_TEMPLATE.md

**Pull request merged**